tzwhere/tzwhere.py

```diff
@@ -167,11 +191,23 @@ def read_tzworld(input_kind='json', path=None):
         return reader(path)
 
     @staticmethod
+    def _open(filename, mode='r'):
+        filename_zipped = filename + tzwhere.ZIPPED
+        if os.path.isfile(filename):
+            return open(filename, mode)
+        elif os.path.isfile(filename_zipped):
+            return closing(gzip.open(filename_zipped, mode))
+        else:
+            url = "{0}/{1}".format(tzwhere.URL_PREFIX, os.path.basename(filename_zipped))
+            content = requests.get(url).content
+            return closing(gzip.GzipFile(fileobj=BytesIO(content)))
```

Comment:
yeah, the magic was done to make it backwards compatible from the user's perspective. I am totally ok otherwise to provide helper functions which user has to use to create a `file` and then call another method to continue with the rest of the process. That should eliminate the magic going on.